src/grid.rs

```diff
@@ -193,6 +210,53 @@ impl<T> Grid<T> {
     }
 }
 
+impl<'a, T> Iterator for GridIterator<'a, T> {
+    type Item = &'a T;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let last_line = self.grid.num_lines() - Line(1);
+        let last_col = self.grid.num_cols() - Column(1);
+        match self.cur {
+            Point { line, col } if
```

Comment:
I think this match should be refactored to not have the `if` blocks here. Not sure if we can match directly on struct patterns, but should be possible to destructure and match on the tuple values. Eg:

```rust
match (line, col) {
    (Line(0), Column(0)) => /* stuff */,
    // etc
}
```